### PR TITLE
Release v0.4.1

### PR DESCRIPTION
**Pull request by relyt29**
## Purpose of Changes and their Description

This is a small release to fix an incomplete migration in 0.4.0, as well as a new mint module endpoint to stimulate the recalculation of the inflation target at any time.

## Link(s) to Ticket(s) or Issue(s) resolved by this PR

See Changelog.txt

### CODE REVIEW

File: x/emissions/migrations/v4/migrate_test.go [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+			EpochLastEnded:         0,
+			EpochLength:            blockHeightEnded3,
+			GroundTruthLag:         30,
+			PNorm:                  alloraMath.NewDecFromInt64(3),
+			AlphaRegret:            alloraMath.MustNewDecFromString("0.1"),
+			AllowNegative:          false,
+			Epsilon:                alloraMath.MustNewDecFromString("0.01"),
+			InitialRegret:          alloraMath.MustNewDecFromString("11"),
+			WorkerSubmissionWindow: 130,
+		},
```
Comment (xmariachi) [inline x/emissions/migrations/v4/migrate_test.go:409]: Please add cases for when values are NaN. In particular `initial_regret`. 
(our topic 5 currently has this value).

File: x/emissions/migrations/v4/migrate.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	cdc := emissionsKeeper.GetBinaryCodec()
+
+	ctx.Logger().Info("MIGRATING STORE FROM VERSION 3 TO VERSION 4")
+	if err := MigrateTopics(ctx, store, cdc, emissionsKeeper); err != nil {
+		ctx.Logger().Error("ERROR INVOKING MIGRATION HANDLER MigrateTopics() FROM VERSION 3 TO VERSION 4")
+		return err
+	}
+
+	ctx.Logger().Info("MIGRATING STORE FROM VERSION 3 TO VERSION 4")
+	ResetMapsWithNonNumericValues(store, cdc)
```
Comment (xmariachi) [inline x/emissions/migrations/v4/migrate.go:32]: why are we resetting all maps here :)

Comment (relyt29) [reply x/emissions/migrations/v4/migrate.go:32]: #583

File: x/emissions/Makefile [hunk 3]
```diff
@@ -32,7 +32,7 @@ proto-gen:
 	@$(protoImage) sh ./scripts/protocgen.sh
 	@go mod tidy
 
-proto-format:
```
Comment (kpeluso) [inline x/emissions/Makefile:35]: should be on x/mint too?

File: x/emissions/types/msg_create_topic.go [hunk 4]
```diff
@@ -37,6 +37,25 @@ func (msg *MsgCreateNewTopic) Validate() error {
...
+	if !isAlloraDecBetweenZeroAndOneInclusive(msg.MeritSortitionAlpha) {
+		return errors.Wrap(sdkerrors.ErrInvalidRequest, "merit sortition alpha must be between 0 and 1 inclusive")
+	}
+	if !isAlloraDecBetweenZeroAndOneInclusive(msg.ActiveInfererQuantile) {
+		return errors.Wrap(sdkerrors.ErrInvalidRequest, "active inferer quantile must be between 0 and 1 inclusive")
+	}
+	if !isAlloraDecBetweenZeroAndOneInclusive(msg.ActiveForecasterQuantile) {
+		return errors.Wrap(sdkerrors.ErrInvalidRequest, "active forecaster quantile must be between 0 and 1 inclusive")
+	}
+	if !isAlloraDecBetweenZeroAndOneInclusive(msg.ActiveReputerQuantile) {
```
Comment (kpeluso) [inline x/emissions/types/msg_create_topic.go:56]: quantiles technically within [0,1)

should see what happens in the quantile-computing function when the quantile arg = 1

Comment (relyt29) [reply x/emissions/types/msg_create_topic.go:56]: you can't have a 100th percentile / quantile?

File: x/emissions/Makefile (hunk 3 above)
Comment (relyt29) [reply x/emissions/Makefile:35]: #584

File: x/emissions/types/msg_create_topic.go (hunk 4 above)
Comment (kpeluso) [reply x/emissions/types/msg_create_topic.go:56]: yeah doesn't technically exist

File: x/emissions/Makefile (hunk 3 above)
Comment (kpeluso) [reply x/emissions/Makefile:35]: im just saying proto-fmt not in that one. minor

File: x/emissions/keeper/keeper.go [hunk 5]
```diff
@@ -210,13 +213,20 @@ func NewKeeper(
...
+		config.MaxMetadataLen = defaultConfig.MaxMetadataLen
+	}
+
 	sb := collections.NewSchemaBuilder(storeService)
 	k := Keeper{
 		cdc:                                      cdc,
 		storeService:                             storeService,
 		addressCodec:                             addressCodec,
 		feeCollectorName:                         feeCollectorName,
+		Config:                                   config,
```
Comment (relyt29) [inline x/emissions/keeper/keeper.go:229]: @guilherme-brandao is this really the best way to do this, seems like a big change to store some new config object on the keeper, we should have to update InitGenesis as well, etc

Comment (relyt29) [reply x/emissions/keeper/keeper.go:229]: I think moduleParams is less awkward way to go...

i had thought we were sourcing something from cosmos-sdk

File: x/emissions/migrations/v4/migrate_test.go (hunk 1 above)
Comment (relyt29) [reply x/emissions/migrations/v4/migrate_test.go:411]: #587

File: x/emissions/migrations/v4/migrate_test.go [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+
+	topicStore := prefix.NewStore(store, types.TopicsKey)
+	for i, oldTopic := range oldTopics {
+		bz, err := proto.Marshal(&oldTopic)
+		s.Require().NoError(err)
+
+		topicStore.Set([]byte("testKey"+strconv.Itoa(i+1)), bz)
+	}
+
+	err = v3.MigrateTopics(s.ctx, store, cdc, *s.emissionsKeeper)
```
Comment (RedBird96) [inline x/emissions/migrations/v4/migrate_test.go:435]: Shouldn't we test `v4.MigrateTopics` at here?

Comment (relyt29) [reply x/emissions/migrations/v4/migrate_test.go:435]: fixed in #587

File: x/emissions/keeper/keeper.go (hunk 5 above)
Comment (relyt29) [reply x/emissions/keeper/keeper.go:229]: fixed in #588

**Comment by relyt29** [comment]
in order to name the branch correctly I will have to close this PR and create a new one (with the same state) for 0.5.0

**Comment by relyt29** [comment]
freezing this branch, locking it, and closing the PR. 

we're renaming the v0.4.1 release to v0.5.0 as it is now a breaking change for off-chain clients.

**Pull request closed without merging**